### PR TITLE
Update PPP and DG

### PR DESCRIPTION
**Pull request by raydenlim**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -241,16 +241,90 @@ Below is a class diagram describing the implementation of `Task` and its respect
...
+
+**Alternative 3 (current choice):** Use both Strings and Structured Objects for Graded Test Calculators.
+- Pros:
+  * Type Safety Utilization: Ensures type safety with objects and allows the use of strings for flexibility. We can take advantage of the strings to enable the use of `default` values.
+  * Robust Testing: Better bug identification, as the two constructors check each other.
+- Cons:
+  * Extra caution is needed to ensure that both these constructors are compatible with one another. 
+
+
+**Aspect 3: How to store graded test scores for individuals**
```
Comment (ldinghan) [inline docs/DeveloperGuide.md]: I believe this line should end with a colon, to standardise with Aspect 1 and Aspect 2

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -241,16 +241,90 @@ Below is a class diagram describing the implementation of `Task` and its respect
...
+
+**Alternative 1(current choice):** Piggy back on the Person Object
```
Comment (ldinghan) [inline docs/DeveloperGuide.md]: Missing a full stop

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -241,16 +241,90 @@ Below is a class diagram describing the implementation of `Task` and its respect
...
-**Alternative 2:** Multiple constructors for `GradedTest`.
-- Pros: Provides flexibility in how users can create a `GradedTest` object.
-- Cons: Requires additional parsing. Different constructor will give different error messages.
+- Pros:
+  * Flexibility: This choice offers flexibility in representing the graded tests, allowing for customization (e.g. usage of default value, `-`).
+  * Clarity: It will be 
+- Cons:
+  * Parsing Overhead: It requires additional parsing to generate statistics of how the Avenger's students are doing (e.g average,
+    highest, lowest score).
+  * Documentation Issues: Since the format for editing graded test follows: `gt/RA1:0 | RA2:0 | MidTerms:0 | Finals:0 | PE:0`
```
Comment (ldinghan) [inline docs/DeveloperGuide.md]: Missing a full stop

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -241,16 +241,90 @@ Below is a class diagram describing the implementation of `Task` and its respect
...
+
+**Alternative 2 :** Using floats for graded test score.
+- Pros:
+  * Numeric Operations: Easier and more efficient numeric operations without the need for extensive parsing.
+- Cons: 
+  * Lesser Flexibility and Customisation: Doesn't allow for special characters
```
Comment (ldinghan) [inline docs/DeveloperGuide.md]: Missing a full stop

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -241,16 +241,90 @@ Below is a class diagram describing the implementation of `Task` and its respect
...
+
+**Alternative 1(current choice):** Piggy back on the Person Object
+- Pros:
+  * Coherence: Graded test information is closely associated with the person(student) it belongs to.
+  * Simplicity: Simplifies the overall structure by leveraging the existing Person object.
+- Cons:
+  * Immutability: Any updates to graded test scores to the Person Objects may cause issues down stream on methods referencing to the old Person object.
+  * Complex Handling of Person: More logic will have to be implemented to update the references of the new Person object.
+
+**Alternative 2:** Isolate the GradedTest Object as its own
```
Comment (ldinghan) [inline docs/DeveloperGuide.md]: Missing a full stop

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -241,16 +241,90 @@ Below is a class diagram describing the implementation of `Task` and its respect
...
+  * Complex Handling of Person: More logic will have to be implemented to update the references of the new Person object.
+
+**Alternative 2:** Isolate the GradedTest Object as its own 
+- Pros:
+  * Easier to debug: Since the GradedTest Object is isolated.
+- Cons:
+  * Immutability: Any updates to Person Object will result in the GradedTest Object referencing the deleted person.
+
+
+**Aspect 4: How to initialise a graded test instance for users**
```
Comment (ldinghan) [inline docs/DeveloperGuide.md]: Same as for Aspect 3, should end with a colon 

OR alternatively remove the colon at the end of the lines for Aspects 1 and 2 for standardisation

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -241,16 +241,90 @@ Below is a class diagram describing the implementation of `Task` and its respect
...
+
+**Alternative 1:** Piggy back on the Person methods via the `add` and `edit` function.
+- Pros:
+  * Convenience: Building on Aspect 2, it enables users to easily set graded test scores to the default values with `gt/default`
```
Comment (ldinghan) [inline docs/DeveloperGuide.md]: Missing a full stop

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -241,16 +241,90 @@ Below is a class diagram describing the implementation of `Task` and its respect
...
+  * Object safety: Building on Aspect 3, since all objects are `private final` and hence immutable, any edits/updates to the person's object will not cause issues to the graded test, as the Person parser logic will handle the new Person object creation.
+- Cons: 
+  * Annoying: The strict GradedTest string constructor does not allow the dynamic edits/updates of scores. (i.e If a person just wants to update a specific test score, they will still have to type out all the scores as input).
+
+**Alternative 2:** Make a separate function `editgradedtest` to update graded test scores
```
Comment (ldinghan) [inline docs/DeveloperGuide.md]: Missing a full stop

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -241,16 +241,90 @@ Below is a class diagram describing the implementation of `Task` and its respect
...
+- Pros:
+  * Dynamic: It enables users to dynamically update their graded test scores without having to use the String format.
+- Cons:
+  * Command Prefixes: Users will have to get used to the command prefix. 
+
+**Alternative 3 (current choice):** Implement both Alternative 1 and 2.
```
Comment (ldinghan) [inline docs/DeveloperGuide.md]: Should be "Alternatives 1 and 2" instead?

**Review by ldinghan — COMMENTED** [review]
LGTM other than some nits for standardisation purposes

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -241,16 +241,90 @@ Below is a class diagram describing the implementation of `Task` and its respect
...
-- Pros: This choice offers flexibility in representing graded tests (e.g Default value of `-`).
-- Cons: It requires additional parsing to generate statistics of how the Avenger's students are doing (e.g average,
-  highest, lowest score).
-
-**Alternative 2:** Multiple constructors for `GradedTest`.
-- Pros: Provides flexibility in how users can create a `GradedTest` object.
-- Cons: Requires additional parsing. Different constructor will give different error messages.
+- Pros:
+  * Flexibility: This choice offers flexibility in representing the graded tests, allowing for customization (e.g. usage of default value, `-`).
+  * Clarity: It will be
```
Comment (wesho1107) [inline docs/DeveloperGuide.md]: Incomplete sentence.

**Review by wesho1107 — COMMENTED** [review]
Other than a small nit, LGTM. Do read and change according to other reviews.

File: docs/DeveloperGuide.md (hunk 10 above)
Comment (raydenlim) [reply docs/DeveloperGuide.md]: eh woops

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -230,22 +230,102 @@ The GradedTest component is responsible for tracking and managing graded test sc
...
-**Alternative 2:** Multiple constructors for `GradedTest`.
-- Pros: Provides flexibility in how users can create a `GradedTest` object.
-- Cons: Requires additional parsing. Different constructor will give different error messages.
-
+- Pros:
+  * Flexibility: This choice offers flexibility in representing the graded tests, allowing for customization (e.g. usage of default value, `-`).
+  * Clarity: The string format of `gt/RA1:<SCORE> | RA2:<SCORE> | MidTerms:<SCORE> | Finals:<SCORE> | PE:<SCORE>` is self-explanatory.
+- Cons:
+  * Parsing Overhead: It requires additional parsing to generate statistics of how the Avenger's students are doing (e.g average, highest, lowest score).
+  * Documentation Issues: The usage of `|` causes conflict with the table notation. Hence, more work is needed to get around this issue in markdown.
```
Comment (Respirayson) [inline docs/DeveloperGuide.md:246]: idt this is a valid con HAHAHA

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -230,22 +230,102 @@ The GradedTest component is responsible for tracking and managing graded test sc
...
+  * Complexity: Since all the scores are in a single string, additional parsing will be needed to implement getters and setters.
+  * Learning Curve: Users might need to adapt to a more structured input format, potentially increasing the learning curve.
+  
+**Alternative 2:** Using Structured Object for Graded Test Constructors.
+- Pros:
+  * Type Safety: A structured object ensures type safety, reducing the risk of runtime errors related to data types.
+  * Clarity and Readability: A well-defined object provides clarity and improves the readability of the code, making it easier to understand.
+- Cons:
+  * Less Flexibility: All 5 fields, `ra1, ra2, midterms, finals, pe` needs to be present at all times. Need to create 5 objects for 1 graded test.
+  * Logging/Documentation: Additional parsing is needed to properly store these fields in the json file properly.
```
Comment (Respirayson) [inline docs/DeveloperGuide.md]: needed to store these fields in the json file properly. Repeated properly

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -230,22 +230,102 @@ The GradedTest component is responsible for tracking and managing graded test sc
...
+
+**Alternative 3 (current choice):** Use both Strings and Structured Objects for Graded Test Calculators.
+- Pros:
+  * Type Safety Utilization: Ensures type safety with objects and allows the use of strings for flexibility. We can take advantage of the strings to enable the use of `default` values.
+  * Robust Testing: Better bug identification, as the two constructors check each other.
```
Comment (Respirayson) [inline docs/DeveloperGuide.md]: Maybe need to elaborate on how the constructors check each other?

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -230,22 +230,102 @@ The GradedTest component is responsible for tracking and managing graded test sc
...
+- Cons:
+  * Extra caution is needed to ensure that both these constructors are compatible with one another. 
+
+<br>
+
+**Aspect 3: How to store graded test scores for individuals:**
+
+**Alternative 1(current choice):** Piggy back on the Person Object.
```
Comment (Respirayson) [inline docs/DeveloperGuide.md]: Phrasing too casual I feel

File: docs/DeveloperGuide.md [hunk 15]
```diff
@@ -230,22 +230,102 @@ The GradedTest component is responsible for tracking and managing graded test sc
...
+- Pros:
+  * Easier to debug: Since the GradedTest Object is isolated.
+- Cons:
+  * Immutability: Any updates to Person Object will result in the GradedTest Object referencing the deleted person.
+
+<br>
+
+**Aspect 4: How to initialise a graded test instance for users:**
+
+**Alternative 1:** Piggy back on the Person methods via the `AddCommand` and `EditCommand` class, with `add` and `edit` respectively.
```
Comment (Respirayson) [inline docs/DeveloperGuide.md]: Similar to above I think piggy back isn't an apt choice of words :(

File: docs/DeveloperGuide.md [hunk 16]
```diff
@@ -230,22 +230,102 @@ The GradedTest component is responsible for tracking and managing graded test sc
...
+- Pros:
+  * Convenience: Building on Aspect 2, it enables users to easily set graded test scores to the default values with `gt/default`.
+  * Object safety: Building on Aspect 3, since all objects are `private final` and hence immutable, any edits/updates to the person's object will not cause issues to the graded test, as the Person parser logic will handle the new Person object creation.
+- Cons: 
+  * Annoying: The strict GradedTest string constructor does not allow the dynamic edits/updates of scores. (i.e If a person just wants to update a specific test score, they will still have to type out all the scores as input).
```
Comment (Respirayson) [inline docs/DeveloperGuide.md]: Perhaps Inputs are not flexible offers more info than just annoying

**Review by Respirayson — APPROVED** [review]
LGTM, just a few small nits

File: docs/DeveloperGuide.md (hunk 11 above)
Comment (raydenlim) [reply docs/DeveloperGuide.md:246]: IT IS VERY PAINFUL FOR ME

**Pull request merged**